torch/_inductor/index_propagation.py

```diff
@@ -322,6 +324,18 @@ def indirect_indexing(
 
             expr = sympy.sympify(index.value.expr)
 
+            # Handle nested indirect indexing, by providing a default
+            # range for indirect symbols. A nested example is:
+            # a = ops.indirect_indexing(...)
+            # b = ops.index_expr(a, ...)
+            # c = ops.indirect_indexing(b, ...)
+            indirect_var_to_default_range = tuple(
+                (sym, self.shape_env._default_unspecified_value_range())
```

Comment:
So, in the pattern above, you are going to hit this function twice. The first one you will not be able to remove the `indirect_indexing`, but you will be able to set proper ranges for it. Then, in the second one, the ranges will already be set, and you will be able to just use those ranges.

The invariant here is that, if you fallback, you will be able to get the result of the fallback, (which is a sympy.Symbol) and add its range as `[-size, size-1]` (as it's not wrapped at this point). Then any call after it should work as expected.
You can even assert that the returned symbol is not in `var_to_range` before adding it.